src/components/ui/InputTag/index.tsx

```diff
@@ -0,0 +1,150 @@
+'use client';
+
+import { useEffect, useRef, useState } from 'react';
+import { clsx } from 'clsx';
+import InputTagAlert from '@/assets/icons/icon-input-tag-alert.svg';
+import InputTagRemove from '@/assets/icons/icon-input-tag-remove.svg';
+
+type InputTagProps = {
+  initialValue?: string;
+  placeholder?: string;
+  onConfirm: (value: string) => void;
+  onRemove?: (value: string) => void;
+  tagErrCheck: (value: string) => boolean;
+  maxLength?: number;
+};
+
+const getVisualLength = (str: string) =>
+  [...str].reduce((len, ch) => len + (ch.charCodeAt(0) > 127 ? 2 : 1), 0);
+
+export const InputTag = ({
+  initialValue = '',
+  placeholder = '#태그',
+  onConfirm,
+  onRemove,
+  tagErrCheck,
+  maxLength = 20,
+}: InputTagProps) => {
+  const [value, setValue] = useState(initialValue);
+  const [isConfirmed, setIsConfirmed] = useState(initialValue.trim().length > 0);
+  const [isFocused, setIsFocused] = useState(false);
+  const [isHovered, setIsHovered] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const spanRef = useRef<HTMLSpanElement>(null);
+  const [inputWidth, setInputWidth] = useState(32);
+
+  useEffect(() => {
+    if (spanRef.current) {
+      const width = spanRef.current.offsetWidth;
+      setInputWidth(Math.min(width + 16, 200));
+    }
+  }, [value, placeholder]);
+
+  const confirmInput = () => {
+    const trimmed = value.trim();
+    if (!trimmed) return;
+
+    const hasError = tagErrCheck(trimmed);
+    setIsError(hasError);
+    setIsConfirmed(true);
+    setIsFocused(false);
+    onConfirm(trimmed);
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      e.preventDefault();
+      confirmInput();
+    }
+  };
+
+  const handleBlur = () => {
+    confirmInput();
+  };
+
+  const enterEditMode = () => {
+    setIsFocused(true);
+    setIsConfirmed(false);
+  };
+
+  const baseStyle =
+    'group flex items-center gap-1 px-2 py-1 text-sm font-medium rounded-[4px] border transition-colors w-fit text-[color:var(--color-layout-grey6)]';
+
+  const statusStyle = clsx({
+    'bg-[color:var(--color-primary-navy1)] border-[color:var(--color-primary-navy3)]':
```

Comment:
p2) 혹시 이 부분

```suggestion
    'bg-primary-navy1 border--primary-navy3':
```

이런 식으로 안 먹히는 걸까요!!

Reply:
수정했읍니다...